langchain4j-azure-ai-search/src/main/java/dev/langchain4j/store/embedding/azure/search/AzureAiSearchEmbeddingStore.java

```diff
@@ -115,21 +117,32 @@ public Builder index(SearchIndex index) {
             return this;
         }
 
+        /**
+         * If no index is provided, set the name of the default index to be used.
```

Comment:
This is just a general comment regarding this approach: I feel the API becomes not that straightforward. When I look at it, I start thinking about what erases what for example.

I suggest considering one more option.
You could reload the same method thus making it more explicit that setting the index name will override setting the index. For example:

```
    public Builder setIndex(SearchIndex index) {
        this.index = index;
        this.indexName = null;  // Clear the index name as the index object takes precedence
        return this;
    }

    public Builder setIndex(String indexName) {
        this.indexName = indexName;
        this.index = null;  // Clear the index object as the index name takes precedence
        return this;
    }
```

Reply:
Same comment as above (this is why I mark this one as resolved, let's have only one discussion). I'm throwing an error, and added more documentation, to be sure people understand what's happening.